Wrappers/Python/cil/optimisation/operators/ProjectionMap.py

```diff
@@ -0,0 +1,62 @@
+from cil.optimisation.operators import LinearOperator
+from cil.framework import BlockGeometry
+
+
+
+class ProjectionMap(LinearOperator):
+
+    r""" Projection Map or Canonical Projection (https://en.wikipedia.org/wiki/Projection_(mathematics))
+    
+    Takes an element x = (x_{0},\dots,x_{i},\dots,x_{n}}) from a Cartesian product space X_{1}\times\cdots\times X_{n}\rightarrow X_{i}
+    and projects it to element x_{i} specified by the index i.
+
+    .. math:: \pi_{i}: X_{1}\times\cdots\times X_{n}\rightarrow X_{i}
+
+    .. math:: \pi_{i}(x_{0},\dots,x_{i},\dots,x_{n}) = x_{i}
+
+    The adjoint operation, is defined as 
+
+    .. math:: \pi_{i}^{*}(x_{i}) = (0, \cdots, x_{i}, \cdots, 0)
+
+    :param domain_geometry: The domain of the Projection Map. A BlockGeometry is expected.
+    :type domain_geometry: `BlockGeometry`
+    :param index: Index to project to the corresponding ImageGeometry X_{index}.
+    :type index: int   
+    :return: returns a DataContainer 
+    :rtype: DataContainer    
+
+    """
+
+    
+    def __init__(self, domain_geometry, index, range_geometry=None):
+        
+        self.index = index
+
+        if not isinstance(domain_geometry, BlockGeometry):
+            raise ValueError("BlockGeometry is expected, {} is passed.".format(domain_geometry.__class__.__name__))
+
+        if self.index > len(domain_geometry.geometries):
+            raise ValueError("Index = {} is larger than the total number of geometries = {}".format(index, len(domain_geometry.geometries)))
+
+        if range_geometry is None:
+            range_geometry = domain_geometry.geometries[self.index]
+            
+        super(ProjectionMap, self).__init__(domain_geometry=domain_geometry, 
+                                            range_geometry=range_geometry)   
+        
+    def direct(self,x,out=None):
+                        
+        if out is None:
+            return x[self.index].copy()
+        else:
+            out.fill(x[self.index])
+    
+    def adjoint(self,x, out=None):
+        
+        if out is None:
+            tmp = self.domain_geometry().allocate()
```

Comment:
is `allocate` supposed to return an array of zeros except at `self.index`?
If so could you make it explicit by `allocate(0)`?